### ISSUE TITLE
failed to connect to device

### ISSUE THREAD

**Issue by jywarren** [CMU-CREATE-Lab/node-speck-sensor#1]
I'm using speck-sensor to try to connect (using an adapted version of speck-gateway) and I can see the Speck present, but can't connect to it. Ubuntu 14.04, node 0.10.36. Any ideas?

```
12:26:47.287 [WARN] console - Path names of all connected Specks:
12:26:47.292 [ERROR] speck-gateway - Path names of all connected Specks:
12:26:47.292 [WARN] console -    0003:0010:00
12:26:47.292 [ERROR] speck-gateway -    0003:0010:00
12:26:47.293 [WARN] console - Connecting to the first Speck found...
12:26:47.293 [ERROR] speck-gateway - Connecting to the first Speck found...
[2015-06-05 12:26:47.295] [ERROR] speck-sensor - connect(): connection failed: cannot open device with path 0003:0010:00
[2015-06-05 12:26:47.297] [ERROR] speck-sensor - Speck.create(): failed to connect to speck at path [0003:0010:00] due to error:Error: Connection failed: failed to connect to device at path [0003:0010:00]
12:26:47.297 [WARN] console - Failed to connect to a Speck.  Will retry in 1 minute.
12:26:47.297 [ERROR] speck-gateway - Failed to connect to a Speck.  Will retry in 1 minute.
12:27:47.300 [WARN] console - Path names of all connected Specks:
12:27:47.300 [ERROR] speck-gateway - Path names of all connected Specks:
12:27:47.300 [WARN] console -    0003:0010:00
12:27:47.301 [ERROR] speck-gateway -    0003:0010:00
12:27:47.301 [WARN] console - Connecting to the first Speck found...
12:27:47.301 [ERROR] speck-gateway - Connecting to the first Speck found...
[2015-06-05 12:27:47.303] [ERROR] speck-sensor - connect(): connection failed: cannot open device with path 0003:0010:00
[2015-06-05 12:27:47.303] [ERROR] speck-sensor - Speck.create(): failed to connect to speck at path [0003:0010:00] due to error:Error: Connection failed: failed to connect to device at path [0003:0010:00]
12:27:47.303 [WARN] console - Failed to connect to a Speck.  Will retry in 1 minute.
12:27:47.303 [ERROR] speck-gateway - Failed to connect to a Speck.  Will retry in 1 minute.
```

**Comment by chrisbartley** [comment]
Did you copy the `55-speck.rules` file into the right place?  See the [README](https://github.com/CMU-CREATE-Lab/node-speck-sensor/tree/master/etc/linux) about it.  I can't remember what Linux does if you haven't done so--does it not see the Speck at all (probably), or simply not connect?  Anyway, worth double-checking.

Also, what kind of Speck do you have?  Black or white?  The node-speck-sensor code is older, and doesn't (yet?) have support for the white, wifi Specks. At a minimum, the USB vendor and product IDs are different for the wifi Specks (VID is `0x2B6E` (or, possibly, but doubtfully, `0x2354`) and PID is `0x3335`).  But the protocol has changed and been extended quite a bit.  Note that the VID and PID are specified in both the `55-speck.rules` file, and in at the top of the `index.js` in node-speck-sensor.

The best code I have that can talk to both black and white specks is in the [speck-chrome-app](https://github.com/CMU-CREATE-Lab/speck-chrome-app) repository.  It's a Chrome app, so some details would need to be changed to move to Node.js, but it shouldn't be too bad.  I'll do so eventually, but just haven't had the time yet.

**Comment by jywarren** [comment]
Ah, I found that you have to run the command with sudo privileges. It's working now! https://github.com/jywarren/opk-speck-cli

It's a black Speck. Thanks for your quick reply!

**Comment by chrisbartley** [comment]
Ah, right, yes!  Sorry that I forgot to add that in the docs.

Fixed in commit 3122643d
